src/AaveV3StrategicAssets_20230606/VeTokenManager.sol

```diff
@@ -0,0 +1,219 @@
+  }
+
+  /// @notice Sells boost in order to accrue rewards
+  /// @dev Registers a new user, creates a BoostOffer with the given parameters
+  /// @param pricePerVote Price of 1 vote per second (in wei)
+  /// @param maxDurationWeeks Maximum duration (in weeks) that a Boost can last when taken from this Offer
+  /// @param expiryTime Timestamp when this Offer is no longer valid
+  /// @param minPerc Minimum percent of users voting token balance to buy for a Boost (in BPS)
+  /// @param maxPerc Maximum percent of users total voting token balance available to delegate (in BPS)
+  /// @param useAdvicePrice True to use the advice Price instead of the given pricePerVote
+  function sellBoost(
+    uint256 pricePerVote,
+    uint64 maxDurationWeeks,
+    uint64 expiryTime,
+    uint16 minPerc,
+    uint16 maxPerc,
+    bool useAdvicePrice
+  ) external onlyOwnerOrGuardian {
+    IERC20(IWardenBoost(WARDEN_VE_BAL).delegationBoost()).approve(
+      WARDEN_VE_BAL,
+      type(uint256).max // Per Boost docs, approve uint256 max
+    );
+    IWardenBoost(WARDEN_VE_BAL).register(
+      pricePerVote,
+      maxDurationWeeks,
+      expiryTime,
+      minPerc,
+      maxPerc,
+      useAdvicePrice
+    );
+    emit SellBoost(pricePerVote, maxDurationWeeks, expiryTime, minPerc, maxPerc, useAdvicePrice);
+  }
+
+  /// @notice Update an existing boost offer
+  /// @param pricePerVote Price of 1 vote per second (in wei)
+  /// @param maxDurationWeeks Maximum duration (in weeks) that a Boost can last when taken from this Offer
+  /// @param expiryTime Timestamp when this Offer is no longer valid
+  /// @param minPerc Minimum percent of users voting token balance to buy for a Boost (in BPS)
+  /// @param maxPerc Maximum percent of users total voting token balance available to delegate (in BPS)
+  /// @param useAdvicePrice True to use the advice Price instead of the given pricePerVote
+  function updateBoostOffer(
+    uint256 pricePerVote,
+    uint64 maxDurationWeeks,
+    uint64 expiryTime,
+    uint16 minPerc,
+    uint16 maxPerc,
+    bool useAdvicePrice
+  ) external onlyOwnerOrGuardian {
+    IWardenBoost(WARDEN_VE_BAL).updateOffer(
+      pricePerVote,
+      maxDurationWeeks,
+      expiryTime,
+      minPerc,
+      maxPerc,
+      useAdvicePrice
+    );
+    emit SellBoost(pricePerVote, maxDurationWeeks, expiryTime, minPerc, maxPerc, useAdvicePrice);
+  }
+
+  /// @notice Removes existing boost offer
+  function removeBoostOffer() external onlyOwnerOrGuardian {
+    IERC20(IWardenBoost(WARDEN_VE_BAL).delegationBoost()).approve(WARDEN_VE_BAL, 0);
+    IWardenBoost(WARDEN_VE_BAL).quit();
+    emit RemoveBoostOffer();
+  }
+
+  /// @notice Claim fee token rewards accrued by selling veBoost
+  function claimBoostRewards() external onlyOwnerOrGuardian {
+    IWardenBoost(WARDEN_VE_BAL).claim();
+    emit ClaimBoostRewards();
+  }
+
+  /// @notice sets the snapshot space ID
+  /// @param _spaceId The string representation of the spaceId
```

Comment:
The comment says string but the type is bytes32, I think this should explain how the transformation is done. I assume it's just "slice out the length and just return the first 32 characters" but worth checking it's not a hash.

Reply:
added comment on how it's used (it's just a mapping using the bytes32): https://etherscan.io/address/0x469788fE6E9E9681C6ebF3bF78e7Fd26Fc015446#code